crypto/ec/ecp_nistp224.c

```diff
@@ -980,14 +981,26 @@ static void point_add(felem x3, felem y3, felem z3,
 
     /*
      * the formulae are incorrect if the points are equal so we check for
-     * this and do doubling if this happens
+     * this and do doubling if this happens.
+     *
+     * In affine coordinates, (X_1, Y_1) == (X_2, Y_2).
+     *
+     * We use bitwise operations to avoid potential side-channels introduced by
+     * the short-circuiting behaviour of boolean operators.
      */
     x_equal = felem_is_zero(ftmp);
     y_equal = felem_is_zero(ftmp3);
+    /*
+     * The special case of either point being the point at infinity (z{1,2}
+     * is zero), is handled separately later on in this function, so we avoid
+     * jumping to point_double here in those special cases.
+     */
     z1_is_zero = felem_is_zero(z1);
     z2_is_zero = felem_is_zero(z2);
-    /* In affine coordinates, (X_1, Y_1) == (X_2, Y_2) */
-    if (x_equal && y_equal && !z1_is_zero && !z2_is_zero) {
+
+    points_equal = (x_equal & y_equal & (~z1_is_zero) & (~z2_is_zero)) & 1;
```

Comment:
Why `& 1` on the end? I note you don't do something similar for ecp_nistp256.c

Reply:
I missed the comments on this PR somehow, sorry!

So, the return value of `felem_is_zero()` is different in `ecp_nistp224.c` than in the other two, in this file truth is returned as `0x1` in the others as `0xff..ff`.
So for p-224, `~true` becomes `0xff..fe` and `~false = 0xff..ff`, that's why I had to mask out only the very last bit for p-224, while in the other 2 `.c` files `~true = 0x0` and `~false = 0xff..ff` which is fine for evaluation in the `if()`.

Reply:
I think this explanation would make a good comment.